src/truskin-page-skin/web/index.js

```diff
@@ -1,15 +1,14 @@
-(function () {
-    let parentBody = window.parent.document.body;
-    parentBody.style.backgroundImage='url(%%VIEW_URL_UNESC%%[%PageSkinFileName%])';
-    parentBody.classList.add('truskin-page-skin');
-    parentBody.style.backgroundAttachment='fixed';
-    parentBody.style.backgroundPosition='50% 0';
-    parentBody.style.backgroundRepeat='no-repeat no-repeat';
-    parentBody.onclick=function(e) {
-        const target = e.target;
-        if (target.nodeName.toLowerCase() === 'body') {
-            window.open('%%CLICK_URL_UNESC%%[%ClickDestination%]');
-        }
-        return true;
-    };
-})();
+let parentBody = window.parent.document.body;
+parentBody.style.backgroundImage='url(%%VIEW_URL_UNESC%%[%PageSkinFileName%])';
+parentBody.style.backgroundAttachment='fixed';
+parentBody.style.backgroundPosition='50% 0';
+parentBody.style.backgroundRepeat='no-repeat no-repeat';
+parentBody.onclick=function(e) {
+    const target = e.target;
+    if (target.nodeName.toLowerCase() === 'body') {
+        window.open('%%CLICK_URL_UNESC%%[%ClickDestination%]');
+    }
+    return true;
+};
+
+window.top.postMessage('truskinRendered', '*');
```

Comment:
Could we be call `window.parent.postMessage` instead of `window.top.postMessage`? 

`window.top` targets the top level window when there's a nested hierarchy of window objects, whereas window.parent explicitly targets the immediate parent.  We're getting `parentBody` from `window.parent` so I assume this is setup to work with the immediate parent only. You might not get the expected behaviour if for some reason there's a hierarchy of window objects greater than 2, I don't know how likely that is though!

Reply:
Good point. We want it to target the `top` window in cases when there are nested iframes so we have now switched on using `window.top.document.body` instead of `window.parent.document.body`

Reply:
Following this change, I think this can be merged now?